astropy/io/ascii/ipac.py

```diff
@@ -518,7 +518,7 @@ def write(self, table):
                 for line in wrap(
                     str(comment), 80, initial_indent="\\ ", subsequent_indent="\\ "
                 ):
-                    lines.append(line)
+                    lines.append(line)  # noqa: PERF402
```

Comment:
If `table.meta["comments"]` has multiple entries that are too long then do we want a separate warning for each of them? If a single warning is good enough then we could rewrite this code quite aggressively.

Reply:
@taldcroft, could you have a look at this?

Reply:
If we want to keep code behavior exactly as it is then we should rewrite this as 
```python
        # Write meta information
        for comment in map(str, table.meta.get("comments", [])):
            if len(comment) > 78:
                warn(
                    "Comment string > 78 characters was automatically wrapped.",
                    AstropyUserWarning,
                )
            line.extend(
                wrap(comment, 80, initial_indent="\\ ", subsequent_indent="\\ "
            )
```
But if it is good enough to emit a single warning for multiple wrapped comments then we could rewrite this as
```python
        # Write meta information
        raw_comments = table.meta.get("comments", [])
        for comment in raw_comments:
            line.extend(
                wrap(str(comment), 80, initial_indent="\\ ", subsequent_indent="\\ ")
            )
        if len(line) > len(raw_comments):
            warn(
                "Comment string > 78 characters was automatically wrapped.",
                AstropyUserWarning,
            )
```
So either way we can avoid the `noqa` and reduce the number of if-statements and for-loops.

Reply:
@taldcroft, if you have a thought on this I can quickly integrate any changes.

Reply:
Since the warning does not provide any info on what comment had to be wrapped, I don't really see a point in raising them all individually (originally this was raising an error, so only triggered on the first one anyway).
Some summary info should be enough, along the lines of
```python
    f"{len(line) - len(raw_comments)} line(s) of comment string > 78 characters were automatically wrapped.",
```
or counting the number of comments that had been wrapped, but that would require another line of bookkeeping again.

Reply:
Done!

Reply:
https://github.com/astropy/astropy/blob/fab4b2d0017de48eaa83698003a92735f79a5f56/astropy/io/ascii/tests/test_ipac_definitions.py#L98

will have to be updated accordingly.

Reply:
Done.

Reply:
When producing the error message the updated code currently assumes that each overlong comment gets split into two lines. But what if a comment is 200 characters long?

Reply:
That would be a case for my alternative suggestion for bookkeeping of the comments, but my understanding of the message was, whether you have two comments of 150 characters or one of 200, you could in both cases say "2 lines were wrapped". But it's a bit ambiguous.

Reply:
My inclination would be change the text to 
``"comment string lines > 78 characters were automatically wrapped."``

Most code editors have an option for a column at a configurable number of characters. I'm not sure what value is added by saying how many lines were wrapped. Personally if I encountered this warning and  wanted to fix my code I would either examine the string and look for characters past the column line or ask GitHub Copilot to just fix it for me. But happy to implement any agreed-upon solution.

Reply:
This is not parsing code, but comments in the `meta` section of a table, which might have been manually written with an editor, but just as well machine-generated or imported from some other table's header. The original warning(s) indicated whether it's just a single line or many places, so my idea was to preserve that information. But I agree it's not terribly important either way.

Reply:
The rest of the pull request looks good, but I still don't like the phrasing of the updated warning here. If we want to include the line number difference in the warning message a better phrasing would be something like `f"Wrapping comment lines > 78 characters produced {len(lines) - len(raw_comments)} extra line(s)"`

Reply:
Agreed, that's much clearer and unambiguous.

Reply:
Done!